deepchem/models/keras_model.py

```diff
@@ -147,6 +147,7 @@ def __init__(self,
       self.optimizer = optimizer
     self.tensorboard = tensorboard
     self.tensorboard_log_frequency = tensorboard_log_frequency
+    self.stop_training = False
```

Comment:
Instead of using an internal field like this to signal the model to stop, how about having the callback raise a `StopIteration` exception?

Reply:
would this look like:
```
for c in callbacks:
   try:
       c(model, step)
   except StopIteration:
       break
```

`stop_training=False` was also borrowed from Keras's design.

Reply:
Exactly.